vividus/src/main/java/org/vividus/runner/BddStepPrinter.java

```diff
@@ -94,77 +105,105 @@ public static void main(String[] args) throws ParseException, IOException
+        }
+        else
+        {
+            step.setLocation("");
+        }
         return step;
     }
 
-    private static class Step implements Comparable<Step>
+    private static final class Step implements Comparable<Step>
     {
-        private static final String DEPRECATED        = "DEPRECATED              ";
-        private static final String COMPOSITE         = "COMPOSITE IN STEPS FILE ";
-        private static final String EMPTY_PLACEHOLDER = "                        ";
-
-        private String definition;
+        private final String startingWord;
+        private final String pattern;
         private boolean deprecated;
         private boolean compositeInStepsFile;
+        private String location = "";
+
+        private Step(String startingWord, String pattern)
+        {
+            this.startingWord = startingWord;
+            this.pattern = pattern;
+        }
+
+        private String getStartingWord()
+        {
+            return startingWord;
+        }
 
-        private void setDefinition(String definition)
+        private String getPattern()
         {
-            this.definition = definition;
+            return pattern;
         }
 
         private void setDeprecated(boolean deprecated)
         {
             this.deprecated = deprecated;
         }
 
-        public void setCompositeInStepsFile(boolean compositeInStepsFile)
+        private boolean isCompositeInStepsFile()
+        {
+            return compositeInStepsFile;
+        }
+
+        private void setCompositeInStepsFile(boolean compositeInStepsFile)
         {
             this.compositeInStepsFile = compositeInStepsFile;
         }
 
-        @Override
-        public String toString()
+        private String getLocation()
         {
-            return (deprecated ? DEPRECATED : compositeInStepsFile ? COMPOSITE : EMPTY_PLACEHOLDER) + definition;
+            return location;
+        }
+
+        private void setLocation(String location)
+        {
+            this.location = location;
         }
 
         @Override
         public int compareTo(Step anotherStep)
         {
-            return definition.compareTo(anotherStep.definition);
+            return Comparator.comparing(Step::isCompositeInStepsFile)
+                    .thenComparing(Step::getLocation)
+                    .thenComparing(Step::getStartingWord)
+                    .thenComparing(Step::getPattern)
+                    .compare(this, anotherStep);
         }
 
         @Override
-        public boolean equals(Object obj)
+        public boolean equals(Object o)
         {
-            if (obj == this)
+            if (this == o)
             {
                 return true;
             }
-            if (obj == null)
-            {
-                return false;
-            }
-            if (obj.getClass() != this.getClass())
+            if (o == null || getClass() != o.getClass())
```

Comment:
might be replaced with a single instanceof (https://stackoverflow.com/a/26514984)

Reply:
1) `instanceof` accepts subclasses, but it's not desired behavior
2) the provided benchmark looks fragile, +0.22% degradation could be just a deviation, and also we use JDK 11 now